cmd/endpoint.go

```diff
@@ -133,6 +136,14 @@ func NewEndpoint(arg string) (ep Endpoint, e error) {
 		}
 	} else {
 		u = &url.URL{Path: path.Clean(arg)}
+
+		// Check if the arg is a localhost or an IP without a scheme.
+		if strings.Contains(arg, "localhost") || (checkIP4(arg) == true) {
+			if !((u.Scheme == "http" || u.Scheme == "https") &&
+				u.User == nil && u.Opaque == "" && u.ForceQuery == false && u.RawQuery == "" && u.Fragment == "") {
+				return ep, fmt.Errorf("invalid URL endpoint format: missing scheme http or https")
```

Comment:
This if block appears to be incorrect. `u` is being assigned as `u = &url.URL{Path: path.Clean(arg)}` where only the Path part of the URL struct is being set. All [other items](https://golang.org/pkg/net/url/#URL) in the struct are not assigned . So they will have their 0 values by default. In fact `u.Scheme == ""` and that is why the scheme check is not doing anything. The `u` needs to be assigned by parsing the `arg` like `u, err = url.Parse(arg)` or so.